### ISSUE TITLE
训练轮数只有一轮

### ISSUE THREAD

**Issue by chenkejin** [yangjianxin1/Firefly#126]
训练轮数只有一轮就结束了，num_train_epochs完全不生效
我的参数如下：
{
    "output_dir": "./output/firefly-sft-v3",
    "model_name_or_path": "Qwen-7B",
    "deepspeed": "./train_args/ds_z3_config.json",
    "train_file": "./data/wenwen_data_v1.jsonl",
    "num_train_epochs": 5,
    "per_device_train_batch_size": 4,
    "gradient_accumulation_steps": 1,
    "learning_rate": 1e-5,
    "max_seq_length": 512,
    "logging_steps": 300,
    "save_steps": 20000,
    "save_total_limit": 5,
    "lr_scheduler_type": "cosine",
    "warmup_steps": 3000,
    "gradient_checkpointing": false,
    "disable_tqdm": false,
    "optim": "adamw_hf",
    "seed": 42,
    "bf16": true,
    "report_to": "tensorboard",
    "dataloader_num_workers": 5,
    "save_strategy": "steps",
    "weight_decay": 0,
    "max_grad_norm": 1.0,
    "remove_unused_columns": false
}
